### PR TITLE
Per module docs in Filebeat

### PR DESCRIPTION
**Pull request by tsg**
* Each module has to provide a docs.asciidoc file to be included
  in the docs
* Following the MB model, these are collected in the `docs/` folder
  on `make update`
* Structure wise, I added a "Modules" part which has an Overview section
  and then a section for each module

Part of #3159.

### CODE REVIEW

File: filebeat/docs/modules-overview.asciidoc [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+beta[]
+
+Filebeat modules simplify the collection, parsing, and visualization of common
+log formats.
+
+A typical module (say, for the Nginx logs) is composed of one ore
+more filesets (in the case of Nginx, `access` and `error`). A fileset contains
+the following:
+
+* Filebeat prospector configurations, which contain the default paths were to
```
Comment (ruflin) [inline filebeat/docs/modules-overview.asciidoc]: s/were/where

File: filebeat/docs/modules-overview.asciidoc [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+accessible from Filebeat (see the <<filebeat-getting-started,getting started>>
+section). It also assumes that the Ingest Node GeoIP and User Agent plugins are
+installed, which you can do with the following two commands executed in the
+Elasticsearch home path:
+
+[source,shell]
+----------------------------------------------------------------------
+$ sudo bin/elasticsearch-plugin install ingest-geoip
+$ sudo bin/elasticsearch-plugin install ingest-user-agent
+----------------------------------------------------------------------
```
Comment (ruflin) [inline filebeat/docs/modules-overview.asciidoc]: should we mention that ES must be restarted afterwards? (or is this not the case?)

File: filebeat/module/nginx/_meta/docs.asciidoc [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+This module comes with a sample dashboard.
+
+image::./images/kibana-nginx.png[]
+
+[float]
+=== Access logs fileset settings
+
+[float]
+==== var.paths
```
Comment (ruflin) [inline filebeat/module/nginx/_meta/docs.asciidoc]: As this exists for all modules / filesets, should we describe it somewhere global?

Comment (tsg) [inline filebeat/module/nginx/_meta/docs.asciidoc]: It's technically not a requirement to have it, although I suspect all filesets will have it. I'm not sure, I think this creates duplication, on the other hands is more convenient for the user to find all options in one place.

File: filebeat/docs/modules-overview.asciidoc (hunk 2 above)
Comment (tsg) [inline filebeat/docs/modules-overview.asciidoc]: Added a note.

File: filebeat/module/nginx/_meta/docs.asciidoc (hunk 3 above)
Comment (ruflin) [inline filebeat/module/nginx/_meta/docs.asciidoc]: Agree, probably better for the user. It will be trickier to keep it up-to-date.

**Pull request merged**